### PR TITLE
JobScheduler switch using getAllTables to searchTables api

### PR DESCRIPTION
**Pull request by jiang95-dev**
## Summary
JobScheduler is using getAllTables api which can bottlenecks the system and fails the jobs. This PR moves JobScheduler to use searchTables api.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [x] Performance Improvements
- [ ] Code Style
- [x] Refactoring
- [ ] Documentation
- [ ] Tests

Move all usage of getAllTables api inside TablesClient to searchTables api combined with getTable call.

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [x] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

Updated TablesClientTest to reflect the api change.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

**Comment by autumnust** [comment]
@jiang95-dev  is this ready for review now ?

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java [hunk 1]
```diff
@@ -244,15 +244,23 @@ public List<String> getDatabases() {
         Collections.emptyList());
   }
 
-  protected TableMetadata parseGetTableResponse(GetTableResponseBody responseBody) {
-    return TableMetadata.builder()
-        .dbName(responseBody.getDatabaseId())
-        .tableName(responseBody.getTableId())
-        .creator(responseBody.getTableCreator())
-        .build();
+  protected TableMetadata mapTableResponseToTableMetadata(GetTableResponseBody responseBody) {
```
Comment (autumnust) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:247]: is private good enough for this ?

**Review by autumnust — APPROVED** [review]
lg, one minor comment

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java [hunk 2]
```diff
@@ -244,15 +244,23 @@ public List<String> getDatabases() {
...
-        .creator(responseBody.getTableCreator())
-        .build();
+  private TableMetadata mapTableResponseToTableMetadata(GetTableResponseBody responseBody) {
+    TableMetadata metadata =
+        TableMetadata.builder()
+            .dbName(responseBody.getDatabaseId())
+            .tableName(responseBody.getTableId())
+            .build();
+    String creator = getTable(metadata).getTableCreator();
+    return new TableMetadata(creator, responseBody.getDatabaseId(), responseBody.getTableId());
```
Comment (teamurko) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java:254]: better use builder for readability

**Review by teamurko — DISMISSED** [review]

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java (hunk 1 above)
Comment (jiang95-dev) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java]: Yeah updated.

**Review by teamurko — APPROVED** [review]

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java (hunk 2 above)
Comment (jiang95-dev) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java]: Done.

**Pull request merged**